docs/Reference/Task-Statuses.md

```diff
@@ -0,0 +1,309 @@
+# Task Statuses
+
+This document provides an overview of the various task statuses used in our
+system, along with their meanings, usage contexts, and the logic that determines
+these statuses.
+
+---
+
+## Table of Contents
+
+- [Active Task Statuses](#active-task-statuses)
```

Comment:
Undispatched can be inactive, so I'd recommend saying something like "Unfinished task statuses" or move undispatched to its own category.